docs/sphinx/developer/concepts/kernel.rst

```diff
@@ -8,7 +8,34 @@ The SimEng Kernel is split into two classes, ``LinuxProcess`` and ``Linux``.
 LinuxProcess
 ------------
 
-The ``LinuxProcess`` class provides the functionality to process the supplied program. It creates the initial process memory space, including the Executable and Linkable Format (ELF) process image and the stack. The population of the initial stack state is based on the information `here <https://www.win.tue.nl/~aeb/linux/hh/stack-layout.html>`_.
+The ``LinuxProcess`` class provides the functionality to process the supplied program. It creates the initial process memory space, including the Executable and Linkable Format (ELF) process image and the stack. The process memory space created contains all the data required by the program to run.
+
+ELF Parsing
+~~~~~~~~~~~~
+The ELF binaries have a defined structure for 32-bit and 64-bit architectures, all information regarding parsing ELF binaries has been refereced from the `Linux manual page <https://man7.org/linux/man-pages/man5/elf.5.html>`_. The ELF binary is divided into multiple parts. SimEng stores all relevant parts of the ``ELF Binary`` in a ``char[] processImage`` array, which is a private member variable of the ``LinuxProcess`` class.
+
+.. image:: ../../assets/elfstruct.png
+  :alt: ELF Strucutre
+  :align: center
+
+* The `ELF Header` is the First part of `ELF binary`. The `ELF Header` holds information regarding the structure of the `ELF Binary`. SimEng extracts the following from the `ELF Header`:
+
+    * The entry point of binary i.e. the virtual address to which the system first transfers control, thus starting execution.
+    * The file offset of the `ELF Program Headers`.
+    * The size of each entry stored in the `ELF Program Header`.
+* The `ELF Program Header` table is an array of structures, each describing a segment or other information the system needs to prepare the program for execution. An object file segment contains one or more sections. `ELF Program Headers` are meaningful only for the executable and shared object files. SimEng extracts the following from the `ELF Program Headers`:
+
+    * The offset from the beginning of the file at which the first byte of the segment resides.
+    * The number of bytes in the memory image of the segment.
+* SimEng uses these extracted values to loop through all `ELF Program Headers` and looks for the `ELF Program Header` located at largest virtual address range. SimEng uses the largest virtual address and size assosciated with that `ELF Program Header` to create an array called the ``ElfProcessImage``. However, this way SimEng ends up creating a sparse array, in which most of the entries are unused. Internally, SimEng treats these virtual address as physical addresses to index into this large array.
+
+* The segment of an `ELF Program Header` has a type and SimEng only extracts and stores a loadable segment type i.e  the segment which contains initialized data that contributes to the program's memory image. This completes the creation of the ``ElfProcessImage``.
```

Comment:
Possible consider this rewording "The segment referenced by an `ELF Program Header` has a type attribute which explains its contents and how to interpret it. For SimEng, we are only interested in the segments of type ``LOAD`` which specifies a loadable segment. Loadable segments most notably contain the workloads' compiled instructions and initialised data that contributes to the program's memory space."

Reply:
Done.